lib/commands/devices/supported.ts

```diff
@@ -69,45 +69,50 @@ export default class DevicesSupportedCmd extends Command {
 		}),
 		verbose: flags.boolean({
 			char: 'v',
-			description:
-				'add extra columns in the tabular output (ALIASES, ARCH, STATE)',
+			description: 'add extra columns in the tabular output',
 		}),
 	};
 
 	public async run() {
 		const { flags: options } = this.parse<FlagsDef, {}>(DevicesSupportedCmd);
-		const dts = await getBalenaSdk().models.config.getDeviceTypes();
-		let deviceTypes: Array<Partial<SDK.DeviceTypeJson.DeviceType>> = dts.map(
-			(d) => {
-				if (d.aliases && d.aliases.length) {
-					// remove aliases that are equal to the slug
-					d.aliases = d.aliases.filter((alias: string) => alias !== d.slug);
-					if (!options.json) {
-						// stringify the aliases array with commas and spaces
-						d.aliases = [d.aliases.join(', ')];
-					}
-				} else {
-					// ensure it is always an array (for the benefit of JSON output)
-					d.aliases = [];
-				}
-				return d;
-			},
-		);
+		const [dts, configDTs] = await Promise.all([
+			getBalenaSdk().models.deviceType.getAll({
+				$expand: { is_of__cpu_architecture: { $select: 'slug' } },
```

Comment:
@thgreasi, thanks for the following comment on Flowdock:
> https://www.flowdock.com/app/rulemotion/pub/threads/AE_IXh9Xs_I7MYOHS98oLb0xaya
> _Always think of `/v6/device_type` as holding 1000's of records with all DTs of the world. You would probably never want to query the DTs w/o a filter to limit the results, to at least those for which we have a hostApp (and maybe at least one release depending on the use case)._

So it sounds like the query above is incomplete, missing filters. Could you help me figuring out what the filter should be here?

>> Or is it the case that the API's `/device-types/v1/` endpoint is now sharing the same source of truth as the `/v6/device_type` endpoint?
> Kind of. The even though `/device-types/v1/` is getting some of its data from S3, it should not serve anything that's not in `/v6/device_type` (which is populated from the DT contract repo).

I am confused: Should the CLI query `models.deviceType.getAll(... $filter ...)` or not? Was this PR unnecessary? As is, this PR added some 11 device types to the output of `balena devices supported`, which I thought were "good" new entries.

Reply:
@pdcastro as per the FD thread
> You would probably never want to query the DTs w/o a filter to limit the results, to at least those for which we have a hostApp (and maybe at least one release depending on the use case).

This is one of those cases where you care for the DTs that already have a hostApp aka are supported in balenaCloud. Luckily, we had considered this case and have a convenience method for that `models.deviceType.getAllSupported()`
See: https://github.com/balena-io/balena-sdk/blob/a42c9f13f554cebd23f5c0b011b95b043f66e072/lib/models/device-type.ts#L149

Reply:
Thanks @thgreasi 🙏  I have changed the code to use `getAllSupported()` instead of `getAll()`. Consistently to your explanation, `getAll()` returns additional results, which were the "11 extra device types" I had mentioned before.

> `getBalenaSdk().models.deviceType.getAllSupported({`
> To be clear, the above probably does not include those that are discontinued (which afair is represented with `is_archived: true` in the hostApp. If you also think we need the discontinued as well (do you really?)

Indeed. Here's what I've done: I have marked the `--discontinued` and `--verbose` flags deprecated, and I have already added `isV13()` feature switches that will automatically disable the flags in CLI v13. While `isV13()` is false (while in v12), I am still taking the "discontinued" devices from `models.config.getDeviceTypes()`, but already taking the non-discontinued devices from `models.deviceType.getAllSupported()` -- the new source of truth. The `STATE` column and `state` JSON field will automatically disappear when CLI v13 is released, thanks to the feature switches.

The overall result, not particularly exciting, :-) is that the list of the devices printed before and after this PR is unchanged.